src/storage/class-sidebar.php

```diff
@@ -0,0 +1,334 @@
+<?php
+/**
+ * Sidebar_Storage class file
+ *
+ * @package wp-widget-control
+ */
+
+namespace Alley\WP\Widget_Control\Storage;
+
+use Mantle\Contracts\Support\Arrayable;
+
+use function Mantle\Support\Helpers\collect;
+use function Mantle\Support\Helpers\option;
+use function Mantle\Support\Helpers\stringable;
+
+/**
+ * Representation of a single sidebar's widgets.
+ *
+ * Widget IDs are in the format of "widget_id_base-instance_number" and stored
+ * in a single option using the "widget_{id_base}" format. For management of
+ * the widget instances inside the sidebar, use {@see \Alley\WP\Widget_Control\Storage\Widget}.
+ */
+class Sidebar implements Arrayable {
```

Comment:
I recently had to create custom function that did this:

> Get all widgets in specific sidebar, and group them. Group if the previous widget is 'xwz-widget' and the current is not.

Would I be able to do this? I _think_ the `filter` helper method could be used.

Reply:
It could be but you may be better off with processing it yourself with `to_array()` here